tcms/rpc/tests/test_testrun.py

```diff
@@ -106,6 +106,56 @@ def test_should_remove_a_case_run(self):
         self.assertFalse(TestExecution.objects.filter(pk=self.test_execution.pk).exists())
 
 
+class TestGetCases(APITestCase):
+    def _fixture_setup(self):
+        super()._fixture_setup()
+
+        self.plan = TestPlanFactory()
+
+        self.test_case = TestCaseFactory()
+        self.test_case.save()
+        self.plan.add_case(self.test_case)
+
+        self.test_run = TestRunFactory(plan=self.plan)
+
+    def test_get_empty_result_if_no_case_added(self):
+        result = self.rpc_client.TestRun.get_cases(self.test_run.pk)
+        self.assertEqual(0, len(result))
+
+    def test_get_cases(self):
+        self.rpc_client.TestRun.add_case(self.test_run.pk, self.test_case.pk)
+        result = self.rpc_client.TestRun.get_cases(self.test_run.pk)
+        self.assertEqual(1, len(result))
+
+        case = result[0]
+        self.assertEqual(self.test_case.pk, case['id'])
+        self.assertTrue('execution_id' in case.keys())
+        self.assertTrue('status' in case.keys())
+
+
+class TestGetCasesPermission(APIPermissionsTestCase):
+    permission_label = 'testruns.view_testrun'
+
+    def _fixture_setup(self):
+        super()._fixture_setup()
+
+        self.plan = TestPlanFactory()
+
+        self.test_case = TestCaseFactory()
+        self.test_case.save()
+        self.plan.add_case(self.test_case)
+
+        self.test_run = TestRunFactory(plan=self.plan)
+
+    def verify_api_with_permission(self):
+        result = self.rpc_client.TestRun.get_cases(self.test_run.pk)
+        self.assertTrue(isinstance(result, list))
```

Comment:
the result is an empty list here b/c self.test_run doesn't contain any test executions (aka cases). So you don't need most of the variables in the setup method.